### PR TITLE
added release notes log,and link to main page

### PR DESCRIPTION
**Pull request by jaimecasero**
**What this PR does / why we need it**:
add release notes doc,so its published to official doc site
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+//add release-notes with newer on top
+== 8.4.0 version
+=== Additions 
+// New features (whether major or minor) go here
+* Clients passwords are now hashed in DB.
+* Profiles allow arbitrary properties to be saved/retrieved.
+* Added configurable SBC mode.
+
+=== Removals
```
Comment (gsaslis) [inline restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: now that i see the content, i think "Breaking Changes" might make more sense...

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
+= Restcomm Release Notes
+
+These notes group both Community and Telscale changes in a single document.
+Changes marked with (R) sign, means those changes are introduced for the respective
+Telscale commercial product.
+
+//add release-notes with newer on top
+== 8.4.0 version
+=== Additions
```
Comment (gsaslis) [inline restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: perhaps "new features" makes more sense, given this content.

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+
+=== Migration Notes
+// Things to consider during migration from previous release
+* Clients password are considered to be MD5 hashed in DB. Use proper migration 
+scripts as usual (R), to ensure existing clients passwords are properly migrated.
+
+=== External Dependencies Updates
+// any dependencies
+* Console updated to version:
```
Comment (gsaslis) [inline restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: clearly there's a version missing here.

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 1 above)
Comment (jaimecasero) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: yep, i think even for end-user the distiction of bugfixes/newFeatures/breakingchanges is still clear and valuable, we can find diff names so we dont use the SemVer references, but i think is better than  these additions/removals/changes. to me those are not clear, i had hard time finding proper category

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc [hunk 4]
```diff
@@ -0,0 +1,47 @@
+= Restcomm Release Notes
```
Comment (gsaslis) [inline restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: Perhaps "Restcomm Connect Release Notes" ?

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+These notes group both Community and Telscale changes in a single document.
```
Comment (gsaslis) [inline restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: "telscale" => "Commercial" ?

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 3 above)
Comment (jaimecasero) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: yep i was asking fronent squad to get those

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+Changes marked with (R) sign, means those changes are introduced for the respective
```
Comment (gsaslis) [inline restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: "(R)" makes me think "why did they choose (R)? what does it stand for?"

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+// New features (whether major or minor) go here
+* Clients passwords are now hashed in DB.
+* Profiles allow arbitrary properties to be saved/retrieved.
+* Added configurable SBC mode.
+
+=== Removals
+// draws attention to functionality that is getting removed
+* RVD auto-login from Console no longer works after hashed passwords
+
+=== Changes
```
Comment (gsaslis) [inline restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: if these are linked to github issues, i think we should include the links

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+// any difference in functionality
+* Dial Timeout does not cancel task when Callee is busy
+* Configurable inbound/outbound SMPP encoding
+* Race condition on sending BYE to incoming call for a dial fork scenario
+* (R) Fixed SDR event for SMS
+
+
+=== Migration Notes
+// Things to consider during migration from previous release
+* Clients password are considered to be MD5 hashed in DB. Use proper migration
```
Comment (gsaslis) [inline restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: "use proper migration scripts as usual (R)" left me lost.

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 6 above)
Comment (gsaslis) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: how about ($) ? 

;)

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 1 above)
Comment (gvagenas) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: Changed to `New Features`, `Breaking changes`, `Bug Fixes`

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 3 above)
Comment (gvagenas) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: The version number for external projects is still missing... will update when available

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 4 above)
Comment (gvagenas) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: Changed it

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 6 above)
Comment (gvagenas) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: I changed the tag to **commercial** and provided a new section to explain the tags used in the release notes:
* **commercial** available only for RestcommOne product
* **security improvement** changes related to security

Later we can add more tags to be used in order to categorize and enhance changes

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 5 above)
Comment (gvagenas) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: Changed that to `RestcommOne` and also added a new section to explain the tags used in the notes (see below)

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 7 above)
Comment (gvagenas) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: There are no links to github issues

File: restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc (hunk 8 above)
Comment (gvagenas) [reply restcomm/restcomm.docs/sources-asciidoc/src/main/asciidoc/release-notes.adoc]: I cleared this

**Pull request merged**